dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/config/ProgramRuleConfig.java

```diff
@@ -54,34 +54,34 @@
     @Autowired
     private OrganisationUnitGroupService organisationUnitGroupService;
 
-    @Autowired
-    private RuleVariableInMemoryMap inMemoryMap;
-
     @Autowired
     private CurrentUserService currentUserService;
 
     @Autowired
     private ConstantService constantService;
 
-    @Autowired
-    private ProgramRuleService programRuleService;
-    @Autowired
-    private ProgramService programService;
-
-
+    /**
+     * This bean is used in the system when an event is intercepted by
+     * {@link ProgramRuleEngineListener}. Only the notification rule actions are
+     * executed.
+     */
     @Bean( "oldRuleEngine" )
     public ProgramRuleEngine oldRuleEngine( OldImplementableRuleService oldImplementableRuleService )
```

Comment:
not sure if this is mentioned somewhere else, but I think it would be good to explain why it is called "old" and "new". Which one should be used, what are the differences?

Reply:
@luciano-fiandesio "oldRuleEngine" only covers notification rule actions. But "newRuleEngine" covers all types of rule actions.